gossip_test.go

```diff
@@ -22,7 +24,7 @@ var _ gossipConnection = &mockGossipConnection{}
 
 func newTestRouter(name string) *Router {
 	peerName, _ := PeerNameFromString(name)
-	router := NewRouter(Config{}, peerName, "nick", nil)
+	router := NewRouter(Config{}, peerName, "nick", nil, log.New(ioutil.Discard, "", 0))
```

Comment:
@bboreham has been debugging many test failures in this area, so I'd like to get his take.